src/main/groovy/com/ullink/gradle/opencover/OpenCoverPlugin.groovy

```diff
@@ -54,10 +61,38 @@ class OpenCoverPlugin implements Plugin<Project> {
         ret
     }
 
+    File downloadReportGenerator(Project project, String version){
+        def dest = new File(new File(project.gradle.gradleUserHomeDir, 'caches'), 'reportgenerator')
+
+        if (!dest.exists()) {
+            dest.mkdirs()
+        }
+
+        def ret = new File(dest, "reportgenerator-${version}")
+        if (!ret.exists()) {
+            project.logger.info "Downloading & Unpacking Report Generator - version ${version}"
+
+            def tmpFile = File.createTempFile("reportgenerator.zip", null)
+
+            new URL(getReportGeneratorUrlForVersion(version)).withInputStream { i ->
+                tmpFile.withOutputStream {
+                    it << i
+                }
+            }
+            project.ant.unzip(src: tmpFile, dest: ret)
```

Comment:
add a sanity check either after unzip or before return that the file exists.
We could get into the issue where the downloaded zip changes structure at one point in time

Reply:
I don't think we need it here. If it doesn't exists we'll return null, right? 
I agree that maybe we need a check before running the merge command to see if we have the report generator or not.